tracer/src/Datadog.Trace/AppSec/Transports/SecurityTransport.Core.cs

```diff
@@ -0,0 +1,154 @@
+// <copyright file="SecurityTransport.Core.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+#if !NETFRAMEWORK
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Datadog.Trace.AppSec.Waf;
+using Datadog.Trace.Headers;
+using Datadog.Trace.Util.Http;
+using Microsoft.AspNetCore.Http;
+
+namespace Datadog.Trace.AppSec.Transports;
+
+internal partial class SecurityTransport
+{
+    private readonly HttpContext _context;
+
+    internal SecurityTransport(Security security, HttpContext context, Span span)
+        : this(span, new HttpTransport(context), security) => _context = context;
+
+    private bool CanAccessHeaders => true;
+
+    internal IResult ShouldBlockPathParams(IDictionary<string, object> pathParams)
+    {
+        if (_httpTransport.Blocked)
+        {
+            return new NullOkResult();
```

Comment:
I'm not sure I follow the logic of returning "NullOkResult" when we're already blocked? Should the result reflect that we're already in a blocked state?

Reply:
it was mostly to avoid return an IResult? from the try catch block, but I guess it's best to return null and deal with it instead of pretending the result is ok. Fixed in dc48889f4effd361f21335a1b5869221fffc9281